lib/tools/apk-utils.js

```diff
@@ -393,20 +391,24 @@ apkUtilsMethods.install = async function (apk, options = {}) {
   if (options.grantPermissions) {
     const apiLevel = await this.getApiLevel();
     if (apiLevel < 23) {
-      log.debug(`Skipping granting permissions for '${apk}', since ` +
+      log.debug(`Skipping granting permissions for '${appPath}', since ` +
                 `the current API level ${apiLevel} does not support applications ` +
                 `permissions customization`);
     } else {
       additionalArgs.push('-g');
     }
   }
 
+  if (appPath.endsWith(APKS_EXTENSION)) {
```

Comment:
👍 
The installation depends on device environment. Configuring device env, including language preference, is called in first step in `startAndroidSession`. The bundle-install command doesn't install unused languages resources.
https://github.com/appium/appium-android-driver/blob/2bac1672019be24c74fec7b4b5e4301558bfcb66/lib/driver.js#L234

Reply:
in theory we could additionally install these language resources manually or change the setup flow sequence. Which way does look more convenient to you?

Reply:
@KazuCocoa I've changed the way we install apks. Now it is much more customizable.